kedro-datasets/setup.py

```diff
@@ -86,7 +86,8 @@ def _collect_requirements(requires):
     "tensorflow.TensorflowModelDataset": [
         # currently only TensorFlow V2 supported for saving and loading.
         # V1 requires HDF5 and serialises differently
-        "tensorflow~=2.0"
+        "tensorflow-macos~=2.0; platform_system == 'Darwin' and platform_machine == 'arm64'",
+        "tensorflow~=2.0; platform_system != 'Darwin' or platform_machine != 'arm64'",
```

Comment:
https://www.tensorflow.org/install/pip#macos_1

The official guide for MacOS is `pip install tensorflow`? Can you supplement the link where it suggest `tensorflow-macos`?

Reply:
Yep:

> For users of Apple M1 computers, to get native performance, you'll want to follow the instructions found [here](https://developer.apple.com/metal/tensorflow-plugin/).

![image](https://user-images.githubusercontent.com/316517/227211602-bab6e4b5-6962-47be-80b0-65b11d577043.png)

So it should be this guide https://developer.apple.com/metal/tensorflow-plugin/ I'll add a comment